### PR TITLE
kernel: netdev: add AQC107 Ethernet driver

### PR DESCRIPTION
**Pull request by aphorise**
KernelPackage Aquantia AQtion (AQC107) 10Gbps Ethernet NIC such as:
 - TP-Link TX401
 - Asus XG-C100C, ROG Areion 10G NIC, etc.

### CODE REVIEW

File: package/kernel/linux/modules/netdevices.mk [hunk 1]
```diff
@@ -1348,3 +1348,18 @@ define KernelPackage/sfc-falcon/description
 endef
 
 $(eval $(call KernelPackage,sfc-falcon))
+
+define KernelPackage/aqc
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Aquantia AQtion (AQC107) 10Gbps Ethernet NIC
+  DEPENDS:=@PCI_SUPPORT +kmod-ptp +kmod-hwmon-core
+  KCONFIG:= \
```
Comment (hauke) [inline package/kernel/linux/modules/netdevices.mk:1356]: You have to specify the Kconfig symbol and no `\` if there is not additional Kconfig symbol in the next line.

File: package/kernel/linux/modules/netdevices.mk [hunk 2]
```diff
@@ -1348,3 +1348,18 @@ define KernelPackage/sfc-falcon/description
 endef
 
 $(eval $(call KernelPackage,sfc-falcon))
+
+define KernelPackage/aqc
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Aquantia AQtion (AQC107) 10Gbps Ethernet NIC
+  DEPENDS:=@PCI_SUPPORT +kmod-ptp +kmod-hwmon-core
```
Comment (hauke) [inline package/kernel/linux/modules/netdevices.mk:1355]: The CONFIG_AQUANTIA_PHY driver does not depend on any of this. Do you want to pack the `CONFIG_AQTION` driver?

File: package/kernel/linux/modules/netdevices.mk (hunk 1 above)
Comment (aphorise) [reply package/kernel/linux/modules/netdevices.mk]: @hauke thanks - can you please recheck and resolve if correct?

File: package/kernel/linux/modules/netdevices.mk (hunk 2 above)
Comment (aphorise) [reply package/kernel/linux/modules/netdevices.mk:1355]: `+kmod-ptp +kmod-hwmon-core` you mean these? @hauke ?

Comment (aphorise) [reply package/kernel/linux/modules/netdevices.mk]: Please check again.

Comment (aphorise) [reply package/kernel/linux/modules/netdevices.mk]: BTW I'm not sure if I've undertood:
>Do you want to pack the CONFIG_AQTION driver?

Yes indeed I do.

**Comment by Ansuel** [comment]
@aphorise you should squash all the commits and provide SoB, correct commit description and commit title.

You should use something like `kernel: netdev: add AQC107 Ethernet driver` or something like that

File: package/kernel/linux/modules/netdevices.mk (hunk 2 above)
Comment (robimarko) [reply package/kernel/linux/modules/netdevices.mk]: Then you have to pass the config option, I doubt you want the AQR PHY driver here

Comment (aphorise) [reply package/kernel/linux/modules/netdevices.mk]: @robimarko - what am I not getting here?

```shell
find . -iname *.ko | grep aq
  # ./build_dir/target-x86_64_musl/linux-x86_64/linux-5.10.134/drivers/net/phy/aquantia.ko
```

**Comment by aphorise** [comment]
> @aphorise you should squash all the commits and provide SoB, correct commit description and commit title.
> 
> You should use something like `kernel: netdev: add AQC107 Ethernet driver` or something like that

For clairty - can you please claritfy `SoB` - is this with the title or description in mind?

**Comment by hauke** [comment]
Did you build and tried this?
`CONFIG_AQC` does not exist in the Linux kernel.

**Comment by Ansuel** [comment]
Signed off by tag required by every commit
Check merged commit for reference

Il Mar 23 Ago 2022, 20:45 aphorise ***@***.***> ha scritto:

**Comment by aphorise** [comment]
> Did you build and tried this? `CONFIG_AQC` does not exist in the Linux kernel.

hey @hauke - sorry for the guessing I wasnt sure what to set this value as since all prior blocks seemed to have something set.

@Ansuel - how to proceed with squash - I tried rebase approach but it was a mess / challenge. Is it easier if I rebranch from current `master` commit and create a new PR with a single commit? (after I've corrected issues).

File: package/kernel/linux/modules/netdevices.mk (hunk 2 above)
Comment (robimarko) [reply package/kernel/linux/modules/netdevices.mk]: What you want for AQC107 is `CONFIG_AQTION`

**Comment by aphorise** [comment]
Squashed and corrected config as well as tested as per:

```shell
OBUILD='openwrt-x86-64-generic-rootfs.tar' ;
gunzip ${OBUILD}.gz && tar -xvf ${OBUILD} ;
find . -iname aqc && grep -F 'aqc' -r ;
  # ./etc/modules.d/aqc
  # usr/lib/opkg/status:Package: kmod-aqc
  # usr/lib/opkg/info/kmod-aqc.list:/etc/modules.d/aqc
  # usr/lib/opkg/info/kmod-aqc.control:Package: kmod-aqc
  # usr/lib/opkg/info/kmod-aqc.control:SourceName: kmod-aqc
  # grep: openwrt-x86-64-generic-rootfs.tar: binary file matches
  # etc/modules.d/aqc:aqc
```

image

**Comment by aphorise** [comment]
@robimarko @hauke is everything alright with this PR or anything more needed?

**Comment by Ansuel** [comment]
Still no SoB tag, wrong title and wrong commit description. Hope I don't sound rude but these thing are mandatory if you want this merged!

**Comment by robimarko** [comment]
Its still invalid, everything @Ansuel told you plus you are still packaging AQR PHY kmod instead of the correct .ko

**Comment by aphorise** [comment]
No issue - I'll gladly act on all that's requested if I can understand clearly what's asked - which I'm struggling to.

> You should use something like `kernel: netdev: add AQC107 Ethernet driver` or something like that

1. Should I use this for A) `commit` message, B) the **Title** of this PR or C) **both**? SoB is still not clear & I know it's not the swear word nor is the 'tag' reference - please clarify or point me to an example PR that may be used as reference @Ansuel 

> ... plus you are still packaging AQR PHY kmod instead of the correct .ko

2. What is the `correct.ko`? - I only saw `aquantia.ko` I see in the kernel is what I've linked so are you there's another?
In my test with TP-Link TX401 on 22.03-rc6 it did not work - vs this that I've submitted that does - so I dont get how it could be wrong? ... I need more guidance please.

**Comment by robimarko** [comment]
Let's start from scratch, what part of HW do you want to support here?
If its that TX401 then you need Aquantia Aqtion driver which you did manage to select via `CONFIG_AQTION` however you are still packaging the Aquantia PHY driver, not the ethernet driver here.
What you need to include is `drivers/net/ethernet/aquantia/atlantic/atlantic.ko`

**Comment by aphorise** [comment]
Ok let me rebuild and retest anew thanks @robimarko

**Comment by aphorise** [comment]
**SoB** - is an acronym for **Signed-Off-By** - which every PR requies & the [following PR](https://github.com/openwrt/openwrt/commit/4742ba41d0a9b4af23437182162b2ce7c7558d97) may be a good example of what the message should include.

However - I'm still not clear whose e-mail do I include or how the SoB flow works - @Ansuel @robimarko kindly help clarify this last bit.

**Comment by robimarko** [comment]
Its simply your name and email in a standardised format.
git commit -s will do this for you if you set name and email in Git config

**Comment by aphorise** [comment]
> Its simply your name and email in a standardised format. git commit -s will do this for you if you set name and email in Git config

Thanks again @robimarko - do I need to include my complete name (first + last) or can it be my handle (`aphorise`)?
- I often dont use my real name.

**Comment by robimarko** [comment]
It has to be a real full name

**Comment by aphorise** [comment]
Closing in favour of:  #10522.

**Comment by Ansuel** [comment]
Why close this to open the same pr.. Ah whatever let's just work on having this merged.

**Comment by aphorise** [comment]
> Why close this to open the same pr.. Ah whatever let's just work on having this merged.

Sorry - I'm not familiar with the squash / compacting and the merge conflicts that resulted were easier to resolve via seperate PR not that I seem to have everything correctly done?

**Comment by robimarko** [comment]
Are you using Github UI for this or?

Cause, otherwise you never want to close a PR and open a new one, there is no need, do the modifications locally and just force push

**Comment by aphorise** [comment]
Using CLI `git` only 

> Cause, otherwise you never want to close a PR and open a new one, there is no need, do the modifications locally and just force push

Noted & will follow this hereon thanks @robimarko

**Pull request closed without merging**